draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -831,134 +831,207 @@ Instantiating one of these protocols with an elliptic curve group motivates
+cryptographically secure hash function H as the input of an encoding function.
+On the one hand, h(x) is difficult to invert since it is computationally
+intractable to produce an input x that maps to h(x) due to H is pre-image
+resistant.
 On the other hand, the uniformity property is not met as the output of an
 encoding is distinguishable from a random distribution. Hence, using
-hash2curve(alpha) is not sufficient to get uniformity, however it can be used
-as a building block for obtaining a random oracle as is described in {{rom}}.
+h(x) is not sufficient to get uniformity and cannot be used as a random oracle.
+
+Brier et al. {{BCIMRT10}} describe two generic constructions that give a
+hash function approximating a random oracle. Farashahi et al. {{FFSTV13}}
+and Tibouchi and Kim {{TK17}} refine this analysis. In particular, Farashahi
+et al. show that summing two independent evaluations of an injective encoding
+suffices to approximate a random oracle to an elliptic curve. This construction
+is given in {{roadmap}}.
+
+# Roadmap {#roadmap}
+
+A general framework for mapping bit strings into points of an elliptic curve is
+presented. The construction of these mappings relies on three basic functions.
+
+-   The function hash2base, {0, 1}^* -> F, maps arbitrary-length bit strings
+    to elements of a finite field; its implementation is defined in
+    {{hashtobase}}.
+
+-   The function map2curve, F -> E, calculates a point on the elliptic curve E
+    from an element of a finite field. {{encodings}} describes mappings for a
+    range of curve families.
+
+-   The function clear\_cofactor, E -> G, maps any point on the curve E to a
+    subgroup G of the curve. {{cofactor-clearing}} describes methods to perform
+    this operation.
+
+Two top-level functions taking as input a bit string and returning points on
+the curve are introduced. Although these functions have the same interface, the
+distribution of the points they produce are different. The first function is
+called injective encoding and its probability distribution of the points can be
+easily distinguished from a uniform distribution of points on the curve. The
+second function behaves as a hash function since its output is
+indistinguishable from a uniform distribution of the points on the curve.
+
+-   Injective encodings. A bit string is mapped to a point in G, however,
+    the distribution of its output is not uniform.
+
+    ~~~
+    encode2curve(alpha)
+
+    Input: alpha, an arbitrary-length bit string.
+    Output: P, a point in G.
+
+    Steps:
+    1. u = hash2base(alpha, 0)
+    2. Q = map2curve(u)
+    3. P = clear_cofactor(Q)
+    4. return P
+    ~~~
+
+-   Random Oracle. This function yields a uniform distribution and can be used
+    as a random oracle from strings to points on the curve. Where appropriate,
+    the terms random oracle and hash functions for such mappings are used
+    interchangeably.
```

Comment:
If we're going to use the terminology this way, the above sentence should appear earlier. But my preference would be to remove this sentence and just be careful to say "random oracle" in the text.